ext/standard/php_fopen_wrapper.c

```diff
@@ -431,18 +431,44 @@ php_stream * php_stream_url_wrap_php(php_stream_wrapper *wrapper, const char *pa
 }
 /* }}} */
 
+static int php_stream_url_stater(php_stream_wrapper *wrapper, const char *url, int flags, php_stream_statbuf *ssb, php_stream_context *context)
+{
+    if (!strncasecmp(url, "php://", sizeof("php://") - 1)) {
+        url += 6;
+    }
+
+    if (!strncasecmp(url, "temp", sizeof("temp") - 1) || !strcasecmp(url, "memory")) {
+        ssb->sb.st_mode |= S_IRUSR | S_IRGRP | S_IROTH | S_IWUSR | S_IWGRP | S_IWOTH;
+    } else if (
+        !strcasecmp(url, "output")
+        || !strcasecmp(url, "stdout")
+        || !strcasecmp(url, "stderr")
+    ) {
+        ssb->sb.st_mode |= S_IWUSR | S_IWGRP | S_IWOTH;
+    } else if (!strcasecmp(url, "input") || !strcasecmp(url, "stdin")) {
+        ssb->sb.st_mode |= S_IRUSR | S_IRGRP | S_IROTH;
+    }
+    /* else if(!strncasecmp(url, "fd/", sizeof("fd/") - 1)) {
+        TODO implement readable\writable flags for fd/
+    } else if (!strncasecmp(url, "filter/", sizeof("filter/") - 1)) {
+        TODO implement readable\writable flags for filter/
+    } */
```

Comment:
Ignoring those cases seems fine for now. I think we might need to return -1 in that case though, to indicate that this is not supported. Can you also add a test for one of those (fd or filter) just so we see what the behavior will be?